### ISSUE TITLE
UIImage extension scaling change scale property of image to uiscreen default

### ISSUE THREAD

**Issue by neokree** [Alamofire/AlamofireImage#277]
Hello, 
I have an image loaded from assets. This image have a `image.scale` property equal to `1.0`.
Now after I use `.af_imageScaled` or `.af_imageAspectScaled` on that I see that property will change to `2.0`, which I think is the same as issue #89. 

I tried to inspect your code, and I found that you call in either extensions this method: 
`UIGraphicsBeginImageContextWithOptions(size, af_isOpaque, 0.0)` where the third parameter is the scale and if is setted to 0.0 the device main screen  scale is used (from Apple documentation here: https://developer.apple.com/documentation/uikit/1623912-uigraphicsbeginimagecontextwitho?language=objc).

There would be useful if you let the developer set the UIImage scale value.
In my case, I don't need to use the image in a view but I'm scaling it for sending a thumbnail to a server and this method is doubling the image size only because my device screen have a scale of 2.0. 

Please let me know if you would add a scale parameter on af_* extensions.
Thank you for your library and your time

**Comment by cnoon** [comment]
Hi @neokree, thank you for filing this issue. I've just pushed change 61a029d0 into `master` which will go out shortly in AFI 4.0.0. It changes all the `UIImage` extensions to expose a `scale` parameter which defaults to `nil` allowing you to override the scale value used in `UIGraphicsBeginImageContextWithOptions` API. I've also changed the default behavior to no longer use a default of `0.0`, but instead the scale of the image. This should result in much more consistent and expected behavior across the board. It was never really correct to have these functions setting the scale value to `UIScreen.main.scale`.

As a secondary addition, I've added the ability to customize the `ImageResponseSerializer` per download through the `ImageDownloader` and extension APIs on `UIImageView` and `UIButton`. This allows you to change the scale factor of the image download through both APIs which could only be done previously if you used the `Alamofire.Session` directly which no one wants to actually do. This should make the APIs much more flexible to do what you want in all situations regardless of whether you're using an `Alamofire.Session`, `AlamofireImage.ImageDownloader`, or `UIImageView` or `UIButton` extension.

This change will be released as part of AFI 4.0.0 here shortly.

Cheers. 🍻
